src/utils/shallowEqual.ts

```diff
@@ -7,11 +8,21 @@ function is(x: any, y: any) {
   return x !== x && y !== y;
 }
 
-function isObject(obj: any): obj is { [key: string]: any } {
+function isObject(obj: any): obj is AnyObject {
   return obj !== null && typeof obj === "object";
 }
 
-export default function shallowEqual(objA: any, objB: any) {
+function shallowObjectEqual(objA: AnyObject, objB: AnyObject) {
+  const keys = Object.keys(objA);
+
+  if (keys.length !== Object.keys(objB).length) {
+    return false;
+  }
+
+  return keys.every(key => hasOwnProperty.call(objB, key) && is(objA[key], objB[key]));
+}
+
+export function shallowEqualSansFirst(objA: any, objB: any) {
```

Comment:
@jcready Can you explain why this is necessary?

Reply:
I didn't want to re-write all the tests that depend on this functionality and it does seem like `first` is typically only used for pagination which means you generally wouldn't want to invalidate the previous data if it changed.